src-tauri/src/commands/feeds.rs

```diff
@@ -2,26 +2,49 @@ use tauri::State;
 
 use crate::models::settings;
 use crate::models::settings::SettingKey;
+use crate::syndication::Feed as _Feed;
 use crate::{
     models::feeds::{self, Feed, FeedToCreate, FeedToUpdate},
     producer::create_new_items,
-    syndication::fetch_feed_title,
+    syndication::{fetch_content, fetch_feed_title, find_feed_link},
     DbState,
 };
 
+use crate::error::Error;
+
 #[tauri::command]
 pub fn create_feed(db_state: State<DbState>, arg: FeedToCreate) -> Result<String, String> {
+    if arg.link.is_empty() {
+        return Err(Error::EmptyString.to_string());
+    }
+
     let db = db_state.db.lock().unwrap();
     let proxy = settings::read(&db, &SettingKey::Proxy)
         .map(|x| x.value)
         .ok();
-    let title = match fetch_feed_title(&arg.link, proxy.as_deref()) {
+
+    let html_content = fetch_content(&arg.link, proxy.as_deref()).unwrap();
+    let is_feed = html_content.parse::<_Feed>().is_ok();
+
+    let link = if is_feed {
+        arg.link.clone()
+    } else {
+        if let Some(rss_link) = find_feed_link(&html_content).unwrap() {
+            rss_link
+        } else {
+            return Err(Error::InvalidFeedLink(arg.link).to_string());
+        }
+    };
```

Comment:
```suggestion
    } else if let Some(rss_link) = find_feed_link(&html_content).unwrap() {
        rss_link
    } else {
        return Err(Error::InvalidFeedLink(arg.link).to_string());
    };
```
Minor, but Clippy pointed it out: https://rust-lang.github.io/rust-clippy/master/index.html#/collapsible_else_if

Reply:
I admire your review.